src/main/java/uk/ac/susx/tag/classificationframework/clusters/clusteranalysis/ClusterFeatureAnalysis.java

```diff
@@ -555,6 +558,7 @@ public static void main(String[] args) throws IOException, ClassNotFoundExceptio
                 "algorithms"
         );
 
+        // In the Pipeline i chose basic because Arabic Stanford crashes when gets a whole paragraph instead of sentence it throws this error gc overhead limit exceeded after taking long time
```

Comment:
Is the basic tokeniser used for the ar-pipeline.ser that gets saved for use with the surprising phrase detector? 

Was the wikipedia background data tokenised using stanford? 

We should ensure that the surprising phrase detector uses the same tokenisation that was used on the background data, otherwise the output will be weird.

Does the basic tokeniser do anything sensible with arabic text? It's just a custom emoticon and url aware simple regex.

Reply:
Before anything this bit of work was a mystery i was basically guessing the process no one remembers it and it is probably has been done wrong 
1- Yes i saved the basic tokeniser cause Stanford  keep making an error i think mentioned in the comments 
2- I don't think i tokenised it by Stanford 
3- Yeah i think they are both the same i ignored Stanford in this step cause it was overflowing the memory i have all the errors recorded in my machine so i can reproduce it (maybe when you are back we can talk about the steps that needs to be done in this last phase of so i can make it correctly)
4- Surprisingly the basic tokeniser works fine but of course it is limited with spaces and commas and so on but Arabic is more complicated than that so it works fine as long as everything is written in proper spacing but that is not always the case with Arabic

Reply:
I tested the tokeniser and it works fine in the classifier all the options are added but the surprising phrase is not working. I think this is where we should start from.

Reply:
<img width="1168" alt="Screenshot 2020-10-29 at 13 56 32" src="https://user-images.githubusercontent.com/68699166/97583218-97863d00-19ee-11eb-9cb7-74601a8508f8.png">
This is a snap shot of the error i am not sure if that helpful but i guess we can talk about it next week when you are back.